frontend/src/DiagramEditor/DiagramEditor.jsx

```diff
@@ -59,18 +78,119 @@ export class DiagramEditor extends React.Component {
     window.removeEventListener("mousemove", this.onWindowMouseMove);
   }
 
+  generateAuthorId = () => {
+    this.authorId = Math.floor(Math.random() * 1000000000000);
+    console.log("authorId:", this.authorId);
+  };
+
+  initialiseWebSocket = () => {
+    const newSocket = new WebSocket(
+      `wss://${WEBSOCKET_API_ID}.execute-api.eu-west-2.amazonaws.com/Prod`
+    );
+
+    this.socket = newSocket;
+
+    // Connection opened
+    newSocket.addEventListener("open", (event) => {
+      console.log("connection open");
+    });
+
+    // Listen for messages
+    newSocket.addEventListener("message", this.onMessageReceived);
+  };
+
+  onMessageReceived = (event) => {
+    const messageData = JSON.parse(event.data);
+    console.log("message:", messageData);
+    switch (messageData.type) {
+      case "master":
+        this.setState({ isMaster: true });
+        break;
+      case "diagramData":
+        this.setState({ diagramData: messageData.diagramData });
+        break;
+      case "connectionId":
+        this.setState({ connectionId: messageData.connectionId });
+        break;
+      case "change":
+        this.handleChange(messageData);
+        break;
+      default:
+        break;
+    }
+  };
+
+  handleChange = (messageData) => {
+    const newDiagramData = applyChangeToDiagramData({
+      change: messageData.change,
+      diagramData: this.state.diagramData,
+    });
+    this.setState({
+      diagramData: newDiagramData,
+    });
+    if (this.state.isMaster) {
+      this.saveDiagram(newDiagramData);
+    }
+  };
+
+  createRevision = ({ revisionName }) => {
+    const { diagramData } = this.state;
+    this.setState({ isRevisionModalOpen: false });
+
+    axios
+      .post(`${REST_API_URL}/create-revision`, { diagramData, revisionName })
+      .then((response) => {
+        window.location = `/diagrams/${response.data.diagramId}`;
```

Comment:
do we need to change this? I thought you separated revision ID from the diagram ID